### PR TITLE
ROX-15131: Generate test metadata and pass it to kubeburner

### PR DESCRIPTION
**Pull request by rhybrillou**
## Description

The goal of the change is to generate a metadata file holding information regarding test run information, and to hand that information over to kube-burner for later storage in long-term storage along with the test results.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

Manual run of the procedure described in `performance-scale-tests/tests/kube-burner` and check:
- the content of the generated file
- that a call to elastic search is performed when configured

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: performance-scale-tests/tests/metadata-collector/main.go [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+	return b.String()
+}
+
+func (f *MetadataField) getDescription() string {
+	var b strings.Builder
+	b.WriteString(f.Description)
+	if f.AutomaticallyGathered {
+		b.WriteString(" - This field is automatically collected by default")
+	}
+	return b.String()
```
Comment (janisz) [inline performance-scale-tests/tests/metadata-collector/main.go:157]:
```suggestion
if !f.AutomaticallyGathered {
		return f.Description
	} 
	return f.Description + " - This field is automatically collected by default"
```

**Review by janisz — COMMENTED** [review]
Does linter runs on this file?
I think it could be beneficial to wrap errors returned by external packages.
How about adding tests (e.g. get flag name function)

File: performance-scale-tests/tests/metadata-collector/main.go [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+		testWorkloadVersion,
+		totalNodes,
+		workerNodesCount,
+		workerNodesKernelVersion,
+		workerNodesType,
+	}
+
+	AllMetadata = make(map[string]interface{}, len(metadataFields))
+
+	kubeconfigpath = os.Getenv("KUBECONFIG")
```
[the remark this answers is not in the document]
Comment (rhybrillou) [reply performance-scale-tests/tests/metadata-collector/main.go:97]: The variable is a param with defaults.
The defaults are :
- `os.Getenv("KUBECONFIG")`
- `filepath.Join(os.Getenv("HOME"),".kube","config")

I'll put the whole of the input parameters in a dedicated object.

File: performance-scale-tests/tests/metadata-collector/main.go [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+	if writeErr != nil {
+		return writeErr
+	}
+	if writtenBytes < len(bytes) {
+		return fmt.Errorf("wrote %d bytes out of %d", writtenBytes, len(bytes))
+	}
+	return nil
+}
+
+func collectAutomaticData(cmd *cobra.Command) error {
```
[the remark this answers is not in the document]
Comment (rhybrillou) [reply performance-scale-tests/tests/metadata-collector/main.go:240]: I think in that case, the function parameter can be removed completely.

File: performance-scale-tests/tests/metadata-collector/main.go [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+		switch field.FieldType {
+		case integerFieldType:
+			cmd.Flags().Int(field.getFlagName(), field.IntDefault, field.getDescription())
+		case stringFieldType:
+			cmd.Flags().String(field.getFlagName(), field.StrDefault, field.getDescription())
+		}
+	}
+
+	if err := cmd.Execute(); err != nil {
+		os.Exit(1)
```
Comment (janisz) [inline performance-scale-tests/tests/metadata-collector/main.go:189]: how about logging error

File: performance-scale-tests/tests/metadata-collector/main.go [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+		index := cwdLen - i - 1
+		lastChar := currentWorkingDir[index]
+		if !os.IsPathSeparator(lastChar) {
+			continue
+		}
+		gitRepo, repoOpenErr = git.PlainOpen(currentWorkingDir[0:index])
+		if repoOpenErr == nil {
+			break
+		}
+	}
```
Comment (janisz) [inline performance-scale-tests/tests/metadata-collector/main.go:240]: how about extracting it into a function `getGitRoot`?

**Review by janisz — APPROVED** [review]

**Pull request merged**